includes/global-functions.php

```diff
@@ -203,3 +240,348 @@ function distributor_get_registered_data() {
+	if ( ! $media_id ) {
+		return array();
+	}
+
+	// Get the media data.
+	$media = get_post( $media_id );
+	if ( ! $media ) {
+		return array();
+	}
+
+	$media_url = wp_get_attachment_url( $media->ID );
+
+	return array(
+		'title'       => $media->post_title,
+		'caption'     => $media->post_excerpt,
+		'description' => $media->post_content,
+		'alt'         => get_post_meta( $media->ID, '_wp_attachment_image_alt', true ),
+		'url'         => $media_url,
+		'guid'        => $media->guid ?? $media_url,
+	);
+}
+
+/**
+ * Post-distribute callback for media data.
+ * This is the default post-distribute callback for media data, used when the type is set to 'media' in distributor_register_data().
+ *
+ * @param mixed $media_extra_data The extra data to be processed after distribution.
+ * @param mixed $source_media_id  The source media ID.
+ * @param mixed $post_data        The post data.
+ * @return int The ID of the distributed media.
+ */
+function distributor_media_post_distribute_callback( $media_extra_data, $source_media_id, $post_data ) {
+	if ( ! isset( $media_extra_data['url'] ) ) {
+		return $source_media_id;
+	}
+
+	$source_media_url = $media_extra_data['url'];
+
+	// Check if the media already exists on the target site. If yes, return its ID.
+	$media_id = Distributor\Utils\get_attachment_id_by_original_data( $source_media_id, $source_media_url );
+
+	// If media exists, return the media ID.
+	if ( ! empty( $media_id ) ) {
+		return $media_id;
+	}
+
+	$media_id = Distributor\Utils\process_media( $source_media_url, 0, [] );
+	// if the media not processed, return the source media ID.
+	if ( empty( $media_id ) ) {
+		return $source_media_id;
+	}
+
+	// Update the media data.
+	wp_update_post(
+		array(
+			'ID'           => $media_id,
+			'post_title'   => sanitize_text_field( $media_extra_data['title'] ),
+			'post_excerpt' => sanitize_textarea_field( $media_extra_data['caption'] ),
+			'post_content' => sanitize_textarea_field( $media_extra_data['description'] ),
+		)
+	);
+
+	// Update the media meta.
+	if ( ! empty( $media_extra_data['alt'] ) ) {
+		update_post_meta( $media_id, '_wp_attachment_image_alt', sanitize_textarea_field( $media_extra_data['alt'] ) );
+	}
+	update_post_meta( $media_id, 'dt_original_media_id', wp_slash( $source_media_id ) );
+	update_post_meta( $media_id, 'dt_original_media_url', wp_slash( $source_media_url ) );
+
+	// Return the media ID to replace the source reference.
+	return $media_id;
+}
+
+/**
+ * Pre-distribute callback for post data.
+ * This is the default pre-distribute callback for post data, used when the type is set to 'post' in distributor_register_data().
+ *
+ * NOTE:
+ * - The post will be distributed only if it is source post. (i.e. not the post that is distributed from another site).
+ * - The extra data will not be processed to prevent infinite loop.
+ *
+ * @param mixed $post_id The post ID to be processed before distribution.
```

Comment:
There's quite a few docblocks using `mixed` where the type is more defined, so are you able to review the instances of param mixed.


```suggestion
 * @param int $post_id The post ID to be processed before distribution.
```

Reply:
Handled in https://github.com/10up/distributor/pull/1309/commits/2c6994ec7ed58c064d282cc2929f128b2b33a428